test/e2e/validation_test.go

```diff
@@ -503,3 +503,65 @@ func (tc *testContext) validateUpgradeableCondition(expected meta.ConditionStatu
 	}
 	return nil
 }
+
+// runWindowsJob creates a Windows Server Job with the given `name` and execution `command`. In addition, check for
+// errors based on `expectError` and is able to delete the job if specified by `deleteJob`
+func runWindowsJob(name, command string, expectError, deleteJob bool,
+	t *testing.T, testCtx *testContext) *batch.Job {
+	winJob, err := testCtx.createWindowsServerJob(name,
+		[]string{
+			testCtx.getPowerShellExe(), "-Command", command,
+		},
+		nil)
+	require.NoError(t, err, "could not create Windows tester job")
+	if deleteJob {
+		defer testCtx.deleteJob(winJob.Name)
+	}
+	err = testCtx.waitUntilJobSucceeds(winJob.Name)
+	if expectError {
+		assert.Error(t, err, "Windows tester job expected to fail")
+	} else {
+		assert.NoError(t, err, "Windows tester job failed")
+	}
+	return winJob
+}
+
+// getServiceNetworkCIDR returns the cluster serviceNetwork CIDR
+func getServiceNetworkCIDR(t *testing.T, testCtx *testContext) string {
+	clusterNetwork, err := testCtx.client.Config.ConfigV1().Networks().Get(context.TODO(), "cluster", meta.GetOptions{})
+	require.NoError(t, err, "error getting cluster network object")
+	require.NotEmpty(t, clusterNetwork.Spec.ServiceNetwork, "service networks cannot be empty")
+	return clusterNetwork.Spec.ServiceNetwork[0]
+}
+
+// testClusterDNS test the DNS server configuration in a Windows pod
+func testClusterDNS(t *testing.T) {
+	// create context
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+
+	// test DNS Server with forward query
+	runWindowsJob("win-dns-tester-forward-query", "curl.exe -k https://www.openshift.com:443",
```

Comment:
Do we really need to curl and download data to test DNS resolution? Can't we just use [host](https://linux.die.net/man/1/host)?

Reply:
Is [host](https://linux.die.net/man/1/host) a Windows cmdlet? Need to run in a Windows pod.

> Do we really need to curl and download data to test DNS resolution?

Ideally, no. [nslookup](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/nslookup) and  [Resolve-DnsName](https://docs.microsoft.com/en-us/powershell/module/dnsclient/resolve-dnsname?view=windowsserver2019-ps) did not worked inside the Windows pod.

Reply:
> Is host a Windows cmdlet?

No.  

> Need to run in a Windows pod.

I understand that. Windows `host` equivalent was an unwritten assumption

>Ideally, no. nslookup and Resolve-DnsName did not worked inside the Windows pod.

I suggest you figure out why.

Reply:
If `ping.exe` is part of the current images being used, that would be a more lightweight option too.